### PR TITLE
JSUI-3098 Added 'mixed' state to the facet values

### PR DESCRIPTION
**Pull request by jlblais**
The "mixed" state was added to the aria-pressed attribute of the facet values. When a facet is selected/excluded, the other should have aria-pressed="mixed". 

**Button states after the changes :** 

1. When a facet value is nor selected nor excluded (no changes):
- “Select”/”Filter by” button : aria-pressed=”false”
- “Exclude”/”Filter out” button : aria-pressed="false”

2. When a facet value is selected:
- “Select”/”Filter by” button : aria-pressed=”true”
- “Exclude”/”Filter out” button (**Changed**): aria-pressed="mixed"

3. When a facet value is excluded:
- “Select”/”Filter by” button (**Changed**): aria-pressed="mixed"
- “Exclude”/”Filter out” button : aria-pressed="true”

https://coveord.atlassian.net/browse/JSUI-3098

[Deploy](https://dashboard.heroku.com/pipelines/a3535101-5bbf-4a5b-a909-47fcf8c9f149)

### CODE REVIEW

**Comment by ThibodeauJF** [comment]
The Exclude button doesn't need the mixed state, unlike the Select button/checkbox, it's state is not affected wether it's selected or not.

File: src/ui/Facet/ValueElementRenderer.ts [hunk 1]
```diff
@@ -262,6 +262,9 @@ export class ValueElementRenderer {
     el.setAttribute('aria-label', this.ariaLabel);
     el.setAttribute('role', 'button');
     el.setAttribute('aria-pressed', this.ariaPressed);
+
+    const elExclude = this.excludeIcon;
```
Comment (ThibodeauJF) [inline src/ui/Facet/ValueElementRenderer.ts:266]: As per my comment, you can remove that part and the method at line #282 😉

File: unitTests/ui/ValueElementRendererTest.ts [hunk 2]
```diff
@@ -124,9 +124,9 @@ export function ValueElementRendererTest() {
         expect(ariaPressed).toEqual('true');
       });
 
-      it('the aria-pressed attribute of the exclude button is false', () => {
+      it('the aria-pressed attribute of the exclude button is mixed', () => {
```
Comment (ThibodeauJF) [inline unitTests/ui/ValueElementRendererTest.ts:127]: Can remove this

**Review by ThibodeauJF — APPROVED** [review]

File: src/ui/Facet/ValueElementRenderer.ts (hunk 1 above)
Comment (jlblais) [reply src/ui/Facet/ValueElementRenderer.ts:266]: In the JIRA issue, it says the Exclude button should have aria-pressed="mixed".  But I agree the state of the Exclude button isn't affected by the state of the Select button. 

@btaillon Is `aria-pressed="mixed"` really what we want?

Comment (ThibodeauJF) [reply src/ui/Facet/ValueElementRenderer.ts:266]: Hmm that's not how I understood it, sorry. 😅 
The Exclude button always has the state pressed or not, either it excludes or unexcludes.
The Select button selects, deselects, and unexcludes if exclude is pressed.

Comment (btaillon) [reply src/ui/Facet/ValueElementRenderer.ts:266]: Woops, that's my bad.

**Review by btaillon — APPROVED** [review]

**Review by samisayegh — APPROVED** [review]

**Review by olamothe — APPROVED** [review]

**Pull request merged**